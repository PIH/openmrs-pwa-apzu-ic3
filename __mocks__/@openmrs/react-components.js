import types from 'redux-types';
// mock methods

const visitRest = {

  createVisit: (params) => {

    const visit = params.visit;

    if (visit.patient != 'patient_uuid' || visit.visitType != 'visit_type_uuid' || visit.location != 'location_uuid') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const encounterRest = {

  createEncounter: (params) => {

    if (params.encounter.encounterType === 'invalid_encounter_type') {
      throw {
        name: 'Mock Exception',
        message: 'Unable to Submit'
      };
    }
    else {
      return;
    }
  }

};

const BASIC_TYPES = [
  'FETCH_SUCCEEDED',
  'FETCH_PENDING',
  'SET_SUCCEEDED',
  'SET_REQUESTED',
  'SET_FAILED',
  'FETCH_FAILED',
  'FETCH_REQUESTED',
  'SUCCEEDED',
  'PENDING',
  'FAILED',
  'REQUESTED'
];

const LOGIN_TYPES = {
  LOGIN: types('login', BASIC_TYPES),
  LOGIN_LOCATIONS: types('login_locations', BASIC_TYPES),
  LOGOUT: types('logout', BASIC_TYPES)
}

const SESSION_TYPES = types('session', BASIC_TYPES)


export {
  visitRest,
  encounterRest,
  LOGIN_TYPES,
  SESSION_TYPES
};
