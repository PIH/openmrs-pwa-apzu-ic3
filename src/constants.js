// TODO: rename to "config.js"?
// TODO: this will be refactored out into an external config at some point

export const PATIENT_REPRESENTATION = '(uuid,id,display,identifiers:(uuid,identifier,identifierType:(uuid),preferred),person:(uuid,display,gender,age,birthdate,birthdateEstimated,dead,deathDate,causeOfDeath,names,preferredName,addresses,attributes))';
export const ENCOUNTER_REPRESENTATION = '(id,uuid,encounterDatetime,location:(id,uuid,name),encounterType:(id,uuid,name))';
export const VISIT_REPRESENTATION = '(uuid,location:(uuid,display,name),startDatetime,stopDatetime,patient:' + PATIENT_REPRESENTATION + ',encounters:(id,uuid,display,encounterDatetime,location:(id,uuid,name),encounterType:(id,uuid,name),obs:(id,uuid,display)))';

export const ENCOUNTER_TYPES = {
  'BloodPressureEncounterType': {
    uuid: '0C36F6FB-660E-485F-AF04-249579C9EAC9'  // is added to the pihmalawi EncounterTypes
  },
  'NutritionEncounterType': {
    uuid: '6265F6BC-EBC0-4181-91F3-28B70BBFDB61'  // is added to the pihmalawi EncounterTypes
  },
  'CheckInEncounterType': {
    uuid: '55a0d3ea-a4d7-4e88-8f01-5aceb2d3c61b'
  },
  'NurseEvaluationEncounterType': {
    uuid: '1e2f8be8-8ae3-41f1-b908-84f168f26325'
  },
  'HTCEncounterType': {
    uuid: '5B7238C1-23C6-4214-957F-7912A5BE87A9'
  },
  'VLEncounterType': {
    uuid: '9959A261-2122-4AE1-A89D-1CA444B712EA'
  },
};


export const VISIT_TYPES = {
  'ClinicVisitType': {
    uuid: 'f01c54cb-2225-471a-9cd5-d348552c337c'
  }
};

export const IDENTIFIER_TYPES = {
  'ART_IDENTIFIER_TYPE': {
    uuid: '66784d84-977f-11e1-8993-905e29aff6c1'
  },
  'EID_IDENTIFIER_TYPE': {
    uuid: '66786256-977f-11e1-8993-905e29aff6c1'
  },
  'NCD_IDENTIFIER_TYPE': {
    uuid: '11a76c3e-1db8-4d16-9252-9a18b5ed1843'
  }
};

export const LOCATION_TYPES = {
  'UnknownLocation': {
    uuid: '8d6c993e-c2cc-11de-8d13-0010c6dffd0f'
  }
};

export const CONCEPTS = {
  'SystolicBloodPressure' : {
    uuid: '6569bffe-977f-11e1-8993-905e29aff6c1'
  },
  'DiastolicBloodPressure' : {
    uuid: '6569c116-977f-11e1-8993-905e29aff6c1'
  },
  'Height': {
    uuid: '6569c562-977f-11e1-8993-905e29aff6c1'
  },
  'Weight': {
    uuid: '6569c44a-977f-11e1-8993-905e29aff6c1'
  },
  'HTC_RESULTS': {
    uuid: '654b983a-977f-11e1-8993-905e29aff6c1',
    'Reactive': {
      uuid: '6549be7a-977f-11e1-8993-905e29aff6c1',
      name: 'Reactive'
    },
    'Non_Reactive': {
      uuid: '654994c2-977f-11e1-8993-905e29aff6c1',
      name: 'Non-Reactive'
    },
    'Not_Done': {
      uuid: '6557a4ae-977f-11e1-8993-905e29aff6c1',
      name: 'Not performed today'
    }
  },
  'VIRAL_LOAD_TEST_SET': {
    uuid: '83931c6d-0e5a-4302-b8ce-a31175b6475e',
    'Bled': {
      uuid: 'f792f2f9-9c24-4d6e-98fd-caffa8f2383f',
      name: 'Bled'
    },
    'True': {
      uuid: '655e2f90-977f-11e1-8993-905e29aff6c1',
      name: 'Yes'
    },
    'False': {
      uuid: '655e3148-977f-11e1-8993-905e29aff6c1',
      name: 'No'
    },
    'ReasonForNoSample': {
      uuid: '0e447d92-a180-11e8-98d0-529269fb1459',
      name: 'Reason for no sample'
    },
    'UnableToDrawBlood': {
      uuid: '0e447720-a180-11e8-98d0-529269fb1459',
      name: 'Unable to draw blood'
    },
    'PatientRefused': {
      uuid: '6566a4ae-977f-11e1-8993-905e29aff6c1',
      name: 'Patient refused'
    },
    'ReasonForTesting': {
      uuid: '164126AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
      name: 'Reason for testing'
    },
    'Routine': {
      uuid: 'e0821812-955d-11e7-abc4-cec278b6b50a',
      name: 'Routine'
    },
    'Confirmatory': {
      uuid: '65590f06-977f-11e1-8993-905e29aff6c1',
      name: 'Confirmatory'
    },
    'Target': {
      uuid: 'e0821df8-955d-11e7-abc4-cec278b6b50a',
      name: 'Target'
    },
    'LabLocation': {
      uuid: '6fc0ab50-9492-11e7-abc4-cec278b6b50a',
      name: 'Lab Location'
    },
    'NenoGeneXpert': {
      uuid: 'e08214c0-955d-11e7-abc4-cec278b6b50a',
      name: 'Neno GeneXpert'
    },
    'CentralLaboratory': {
      uuid: 'e0820552-955d-11e7-abc4-cec278b6b50a',
      name: 'Central Laboratory'
    },
    'LisungwiGeneXpert': {
      uuid: 'e08212b8-955d-11e7-abc4-cec278b6b50a',
      name: 'Lisungwi GeneXpert'
    }
  }
};

export const NAV_MENU_PAGES = {
  '/' : { display : 'Home', icon : 'home'},
  '/searchPatient' : { display : 'Search Patient', icon : 'search'},
  '/checkin/checkInTabs' : { display : 'Check-In', icon : 'notes-medical'},
  '/screening/bloodPressure/queue' : { display : 'Blood Pressure', icon : 'heart' },
  '/screening/nutrition/queue' : { display : 'Nutrition', icon : 'lemon' },
  '/screening/htc/queue' : { display : 'HTC', icon : 'ribbon' },
  '/screening/vl/tabs' : { display : 'Viral Load', icon : 'ribbon' },
  '/screening/nurse/queue' : { display : 'Nurse', icon : 'user-md' },
};

export const USER_MENU_PAGES = {
  '/user/settings' : { display : 'Settings', icon : 'cog' },
  '/user/settings/password' : { display : 'Change Password', icon : 'key' },
  '/logout' : { display : 'Logout', icon : 'sign-out-alt' },
};

