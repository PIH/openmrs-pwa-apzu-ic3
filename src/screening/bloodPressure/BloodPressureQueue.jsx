import React from "react";
import { connect } from "react-redux";
import { push } from 'connected-react-router';
import {selectors} from "@openmrs/react-components";
import ScreeningTabs from "../ScreeningTabs";
import bloodPressureFilters from './bloodPressureFilters';

// TODO can we figure out a better way to do this without passing dispatch all the way through?

let BloodPressureQueue = props => {

  const rowSelectedActionCreators = [
    () => push('/screening/bloodPressure/form')
  ];

  return (
    <div>

      <ScreeningTabs
        dispatch={ props.dispatch }
        filters={ bloodPressureFilters }
        rowData={ Object.values(props.patients) }
        rowSelectedActionCreators={ rowSelectedActionCreators }
        tabsId="bp-tabs"
        title="Blood Pressure Queue"
      />
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    patients: selectors.getPatientStore(state),
  };
};

export default connect(mapStateToProps)(BloodPressureQueue);
