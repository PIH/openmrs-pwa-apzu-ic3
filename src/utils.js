import dateFns from 'date-fns';
import { patientUtil } from '@openmrs/react-components';
import {
  ENCOUNTER_TYPES, CONCEPTS, MALNUTRITION_LEVEL, EID_RAPID_TEST,
  EID_DNA_PCR
} from "./constants";

const utils = {

  formatTime: (datetime) => {
    return dateFns.format(new Date(datetime), 'h[:]mma');
  },

  formatRestDate: (datetime) => {
    return dateFns.format(datetime, 'YYYY-MM-DDTHH:mm:ss.SSS');
  },
  formatCalendarDate: (datetime) => {
    return dateFns.format(datetime, 'MMMM D, YYYY');
  },
  formatReportRestDate: (datetime) => {
    return dateFns.format(datetime, 'YYYY-MM-DD');
  },
  getTodaysDate: () => {
    return dateFns.format(new Date(), 'YYYY-MM-DD');
  },

  getEndOfYesterday: () => {
    return utils.formatRestDate(dateFns.endOfYesterday());
  },

  getIdentifiersToDisplay(patient, currentLocationPrefix, cccNumber, hccNumber) {
    let identifiers = [], additionalIdentifiers = [];
    const baseIdentifiers = patientUtil.getIdentifiers(patient);
    const hasCCCIdentifier = patientUtil.getIdentifiersAndPreferred(patient, cccNumber);
    const hasHCCIdentifier = patientUtil.getIdentifiersAndPreferred(patient, hccNumber);

    // Check if it had CCC identifier, if only 1 use as default
    if (hasCCCIdentifier.length === 1) {
      identifiers.push(hasCCCIdentifier[0].identifier);
    } else if (hasCCCIdentifier.length > 1) {
      // If more than 1 CCC identifier, use current location identifier
      const getCccCurrentLocation = hasCCCIdentifier.find(identifier => identifier.identifier.match(currentLocationPrefix));
      if (getCccCurrentLocation && currentLocationPrefix) {
        identifiers.push(getCccCurrentLocation[0].identifier);
      } else {
        // If more than 1 CCC identifier and no currentLocation use preferred
        const getCccPreferred = hasCCCIdentifier.find(identifier => identifier.preferred === true);
        if (getCccPreferred) {
          identifiers.push(getCccPreferred[0].identifier);
        } else {
          // if NO preferred CCC identifier, use first one
          identifiers.push(hasCCCIdentifier[0].identifier);
        }
      }
    }

    if (hasHCCIdentifier.length === 1) {
      identifiers.push(hasHCCIdentifier[0].identifier);
    } else if (hasHCCIdentifier.length > 1) {
      const getHccCurrentLocation = hasCCCIdentifier.find(identifier => identifier.identifier.match(currentLocationPrefix));
      if (getHccCurrentLocation && currentLocationPrefix) {
        identifiers.push(getHccCurrentLocation[0].identifier);
      } else {
        const getHccPreferred = hasHCCIdentifier.find(identifier => identifier.preferred === true);
        if (getHccPreferred) {
          identifiers.push(getHccPreferred[0].identifier);
        } else {
          identifiers.push(hasHCCIdentifier[0].identifier);
        }
      }
    }
      
    const patientIdentifiersLength = identifiers.length;
    let uniqueIdentifiers = [...new Set(identifiers.concat(baseIdentifiers))];
    uniqueIdentifiers.splice(0, patientIdentifiersLength);

    if (identifiers.length === 0) {
      identifiers.push(uniqueIdentifiers[0]);
      uniqueIdentifiers.splice(0, 1);
    }
    additionalIdentifiers = uniqueIdentifiers;
    return { identifiers, additionalIdentifiers };
  },
  /* getPatientArtIdentifier: (patient) => {
     return patientUtil.getIdentifier(patient, IDENTIFIER_TYPES.ART_IDENTIFIER_TYPE);
   },
   getPatientEidIdentifier: (patient) => {
     return patientUtil.getIdentifier(patient, IDENTIFIER_TYPES.EID_IDENTIFIER_TYPE);
   },
   getPatientNcdIdentifier: (patient) => {
     return patientUtil.getIdentifier(patient, IDENTIFIER_TYPES.NCD_IDENTIFIER_TYPE);
   },*/
  getPatientIdentifiers: (patient) => {

    const identifiers = patientUtil.getIdentifiers(patient);

    if (identifiers.length > 0) {
      return identifiers.join('<br/>');
    } else {
      return null;
    }


  },
  getPatientCheckedInEncounter: (patient) => {

    let checkedInEncounter = null;
    if (typeof patient.visit !== 'undefined' && typeof patient.visit.encounters !== 'undefined') {
      //filter by CheckIn encounter
      let checkedInEncounters = patient.visit.encounters.filter(encounter =>
        encounter.encounterType.uuid === ENCOUNTER_TYPES.CheckInEncounterType.uuid);

      if (checkedInEncounters.length >  0 ) {
        checkedInEncounters.sort(function (a, b) {
          return +new Date(a.encounterDatetime) - +new Date(b.encounterDatetime);
        });
        checkedInEncounter = checkedInEncounters[0];
      }
    }

    return checkedInEncounter;

  },
  getPatientCheckedInTime: (patient) => {
    return utils.getPatientCheckedInEncounter(patient) !== null ? utils.formatTime((utils.getPatientCheckedInEncounter(patient)).encounterDatetime) : null;
  },

  getPatientCheckedInDate: (patient) => {
    return utils.getPatientCheckedInEncounter(patient) !== null ? utils.formatReportRestDate((utils.getPatientCheckedInEncounter(patient)).encounterDatetime) : null;
  },

  getLastLabTest: (labTests) => {
    let lastLabTest = null;
    if (labTests !== null ) {
      let filteredTests = labTests;
      if (typeof filteredTests !== 'undefined' && filteredTests !== null ) {

        if ( filteredTests.length > 0 ) {
          filteredTests.sort(function (a, b) {
            return +new Date(b.resultDate) - +new Date(a.resultDate);
          });
          lastLabTest = filteredTests[0];
        }
      }
    }

    return lastLabTest;
  },

  findByUuid: (o, uuid) => {
    //early return
    if (o.uuid === uuid) {
      return o;
    }

    let result, p;
    for (p in  o) {
      if ( o.hasOwnProperty(p) && typeof o[p] === 'object') {
        result = utils.findByUuid(o[p], uuid);
        if (result) {
          return result;
        }
      }
    }
    return result;
  },

  getConceptNameByUuid: (uuid) => {
    let conceptName = null;
    if (uuid) {
      let conceptObj = utils.findByUuid(CONCEPTS, uuid);
      if (conceptObj) {
        conceptName = conceptObj.name;
      }
    }
    return conceptName;
  },

  getDefaultEidTestType: (patient) => {

    let eidForm = null;
    if ( (typeof patient.alert !== 'undefined') && (patient.alert !== null) &&  ( Object.keys(patient.alert).length > 0 )) {
      let alertsArray = Object.keys(patient.alert);
      for (let i = 0; i < alertsArray.length; i++) {
        let a = alertsArray[i];
        if ( EID_RAPID_TEST.indexOf(a) >= 0 ) {
          return CONCEPTS.HIV_RAPID_TEST.uuid;
        } else if ( EID_DNA_PCR.indexOf(a) >= 0 ) {
          return CONCEPTS.HIV_DNA_PCR_TEST.uuid;
        }
      }
    }

    return eidForm;
  },

  hasAlert: (alerts, category) => {
    if ((typeof alerts === 'undefined') ||
      (alerts === null) ||
      ( alerts.length < 1)) {
      return false;
    }
    let retValue = false;

    for (let i = 0; i < alerts.length; i++) {
      retValue =  alerts[i].categories.some(a => category.indexOf(a) >= 0);
      if (retValue === true) {
        break;
      }
    }
    return retValue;
  },

  getAdherenceSessionNumber: (obs) => {
    let sessionNumber = null;
    obs.forEach(function(observation) {
      if (observation.concept.uuid === CONCEPTS.ADHERENCE_COUNSELING.AdherenceSession.uuid ) {
        switch (observation.value.uuid) {
          case CONCEPTS.ADHERENCE_COUNSELING.AdherenceSession.FirstSession.uuid:
            sessionNumber = CONCEPTS.ADHERENCE_COUNSELING.AdherenceSession.FirstSession.name;
            break;
          case CONCEPTS.ADHERENCE_COUNSELING.AdherenceSession.SecondSession.uuid:
            sessionNumber = CONCEPTS.ADHERENCE_COUNSELING.AdherenceSession.SecondSession.name;
            break;
          case CONCEPTS.ADHERENCE_COUNSELING.AdherenceSession.ThirdSession.uuid:
            sessionNumber = CONCEPTS.ADHERENCE_COUNSELING.AdherenceSession.ThirdSession.name;
            break;
          default:
            break;
        }
      }
    });
    return sessionNumber;
  },

  getAdherenceCounselor: (obs) => {
    let counselor = null;
    obs.forEach(function(observation)  {
      if (observation.concept.uuid === CONCEPTS.ADHERENCE_COUNSELING.NameOfCounselor.uuid ) {
        counselor = observation.value;
      }
    });
    return counselor;
  },
  
  getPatientCheckOutTime: (patient) => {

    let checkOutTime = null;
    if (typeof patient.visit !== 'undefined' && patient.visit.stopDatetime !== null) {
      checkOutTime = utils.formatTime(patient.visit.stopDatetime);
    }

    return checkOutTime;
  },

  calculateBMI: (weight, height) => {
    let bmi = null;
    if (weight !== null && height !== null ) {
      bmi = (( weight / ( height * height) ) * 10000).toFixed(1);
    }
    return bmi;
  },

  calculateBMIAlert: (bmi) => {
    let alert = MALNUTRITION_LEVEL.default;
    if ( bmi < 16) {
      alert = MALNUTRITION_LEVEL.severe;
    } else if ( bmi >= 16 && bmi < 18.4) {
      alert = MALNUTRITION_LEVEL.moderate;
    } else if ( bmi >=18.4 && bmi < 24.9) {
      alert = MALNUTRITION_LEVEL.normal;
    } else {
      alert = MALNUTRITION_LEVEL.overweight;
    }

    return alert;
  },

  calculateMalnutritionLevel: (bmi,muac, age, pregnant) => {
    let level = null;
    if (typeof bmi !== 'undefined' && bmi !== null && age > 18 && (typeof pregnant === 'undefined' || (typeof pregnant !== 'undefined' && pregnant === CONCEPTS.False.uuid))) {
      //all adults above 18 years excluding pregnant females
      return utils.calculateBMIAlert(bmi);
    } else if (typeof muac !== 'undefined' && muac !== null && age !== null) {
      level = MALNUTRITION_LEVEL.normal;
      switch (true) {
        case (age < 5 ): //MUAC cut-offs: 6-59 months <11.5cm severe and 11.5-12.5 moderate
          if (muac < 11.5) {
            level = MALNUTRITION_LEVEL.severe;
          } else if ( muac < 12.5 ) {
            level = MALNUTRITION_LEVEL.moderate;
          }
          break;
        case (age < 10): //MUAC cut-offs: 5-9 years <13cm severe and 13-14.5 moderate
          if (muac < 13) {
            level = MALNUTRITION_LEVEL.severe;
          } else if ( muac < 14.5 ) {
            level = MALNUTRITION_LEVEL.moderate;
          }
          break;
        case (age < 15): //MUAC cut-offs: 10-14 years <16cm severe and 16-18.5 moderate
          if (muac < 16) {
            level = MALNUTRITION_LEVEL.severe;
          } else if ( muac < 18.5 ) {
            level = MALNUTRITION_LEVEL.moderate;
          }
          break;
        case (age < 18): //MUAC cut-offs: 15-18 years <18.5cm severe and 18.5-21.9cm moderate
          if (muac < 18.5) {
            level = MALNUTRITION_LEVEL.severe;
          } else if ( muac < 21.9 ) {
            level = MALNUTRITION_LEVEL.moderate;
          }
          break;
        case (age >= 18): //MUAC cut-offs: pregnant mothers  <19cm severe and 19-21.9 moderate
          if ( typeof pregnant !== 'undefined' && pregnant === CONCEPTS.True.uuid ){
            if (muac < 19) {
              level = MALNUTRITION_LEVEL.severe;
            } else if ( muac < 22) {
              level = MALNUTRITION_LEVEL.moderate;
            }
          }
          break;
        default:
          break;
      }
    }
    return level;
  }

};

export default utils;
