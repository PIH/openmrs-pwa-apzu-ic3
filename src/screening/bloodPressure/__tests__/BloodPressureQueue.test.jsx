import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { CardList, patientActions } from '@openmrs/react-components';
import BloodPressureQueue from '../BloodPressureQueue';
import { VISIT_REPRESENTATION } from "../../../constants";
import ic3PatientActions from "../../../patient/patientActions";
import utils from "../../../utils";


jest.mock('@openmrs/react-components', () => {
  return {
    patientObjByEncounterTypeFilter: jest.fn(),
    selectors: {
      getPatientStore: jest.fn((state) => ({
        1: {
          name: 'somePatient'
        }
      })),
      isPatientStoreUpdating: jest.fn()
    },
    patientActions: {
      clearSelectedPatient: jest.fn(),
      setSelectedPatient: jest.fn(),
    },
    CardList: () => (
      <div>
        <span>card list component</span>
      </div>
    ),
  };
});

let props, store;
let mountedComponent;

const mockStore = configureMockStore();

const bloodPressureQueue = () => {
  if (!mountedComponent) {
    mountedComponent = mount(
      <Provider store={store}>
        <BloodPressureQueue {...props} />
      </Provider>);
  }
  return mountedComponent;
};

describe('Component: BloodPressureQueue', () => {
  beforeEach(() => {
    props = {
      session: {
        sessionLocation: {
          uuid: 'abc'
        }
      }
    };
    store = mockStore(
      {
        dispatch: {},
        openmrs: {
          session: {
            sessionLocation: {
              uuid: 'abc'
            }
          },
          patients: {
            set: {},
            loading: true,
            selected: null
          }
        },
        router: {
          location: {
            pathname: '/'
          }
        }
      });
    mountedComponent = undefined;
  });

  it('renders properly', () => {
    //expect(toJson(bloodPressureQueue())).toMatchSnapshot();
    expect(bloodPressureQueue().find(CardList).length).toBe(1);
    expect(bloodPressureQueue().find(CardList).get(0).props.rowSelectedActionCreators.length).toBe(3);
    expect(bloodPressureQueue().find(CardList).get(0).props.rowSelectedActionCreators[1]().payload.args[0]).toBe("/screening");
  });

});
