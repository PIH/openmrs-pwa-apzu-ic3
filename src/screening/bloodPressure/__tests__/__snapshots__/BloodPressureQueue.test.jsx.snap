// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BloodPressureQueue renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(BloodPressureQueue)
    session={
      Object {
        "sessionLocation": Object {
          "uuid": "abc",
        },
      }
    }
  >
    <BloodPressureQueue
      dispatch={[Function]}
      patients={Array []}
      session={
        Object {
          "sessionLocation": Object {
            "uuid": "abc",
          },
        }
      }
    >
      <div>
        <Connect(ScreeningQueue)
          dispatch={[Function]}
          filters={
            Array [
              [Function],
              [Function],
            ]
          }
          rowData={Array []}
          rowSelectedActionCreators={
            Array [
              [Function],
            ]
          }
          title="Blood Pressure Queue"
        >
          <ScreeningQueue
            columnDefs={
              Array [
                Object {
                  "field": "uuid",
                  "headerName": "uuid",
                  "hide": true,
                },
                Object {
                  "headerName": "ART",
                  "valueGetter": [Function],
                },
                Object {
                  "headerName": "EID",
                  "valueGetter": [Function],
                },
                Object {
                  "headerName": "NCD",
                  "valueGetter": [Function],
                },
                Object {
                  "field": "name.givenName",
                  "headerName": "Given Name",
                },
                Object {
                  "field": "name.familyName",
                  "headerName": "Family Name",
                },
                Object {
                  "field": "gender",
                  "headerName": "Gender",
                },
                Object {
                  "field": "age",
                  "headerName": "Age",
                },
                Object {
                  "headerName": "Checked-in Time",
                  "valueGetter": [Function],
                },
              ]
            }
            dispatch={[Function]}
            filters={
              Array [
                [Function],
                [Function],
              ]
            }
            rowData={Array []}
            rowSelectedActionCreators={
              Array [
                [Function],
              ]
            }
            session={
              Object {
                "sessionLocation": Object {
                  "uuid": "abc",
                },
              }
            }
            title="Blood Pressure Queue"
          >
            <div>
              <List
                columnDefs={
                  Array [
                    Object {
                      "field": "uuid",
                      "headerName": "uuid",
                      "hide": true,
                    },
                    Object {
                      "headerName": "ART",
                      "valueGetter": [Function],
                    },
                    Object {
                      "headerName": "EID",
                      "valueGetter": [Function],
                    },
                    Object {
                      "headerName": "NCD",
                      "valueGetter": [Function],
                    },
                    Object {
                      "field": "name.givenName",
                      "headerName": "Given Name",
                    },
                    Object {
                      "field": "name.familyName",
                      "headerName": "Family Name",
                    },
                    Object {
                      "field": "gender",
                      "headerName": "Gender",
                    },
                    Object {
                      "field": "age",
                      "headerName": "Age",
                    },
                    Object {
                      "headerName": "Checked-in Time",
                      "valueGetter": [Function],
                    },
                  ]
                }
                delayInterval={10000}
                fetchListActionCreator={[Function]}
                filters={
                  Array [
                    [Function],
                    [Function],
                    [Function],
                  ]
                }
                onMountOtherActionCreators={
                  Array [
                    [Function],
                  ]
                }
                rowData={Array []}
                rowSelectedActionCreators={
                  Array [
                    [Function],
                  ]
                }
                title="Blood Pressure Queue"
              >
                <div>
                  <h3>
                    <Label
                      bsClass="label"
                      bsStyle="default"
                    >
                      <span
                        className="label label-default"
                      >
                        Blood Pressure Queue
                      </span>
                    </Label>
                  </h3>
                  <Connect(DataGrid)
                    columnDefs={
                      Array [
                        Object {
                          "field": "uuid",
                          "headerName": "uuid",
                          "hide": true,
                        },
                        Object {
                          "headerName": "ART",
                          "valueGetter": [Function],
                        },
                        Object {
                          "headerName": "EID",
                          "valueGetter": [Function],
                        },
                        Object {
                          "headerName": "NCD",
                          "valueGetter": [Function],
                        },
                        Object {
                          "field": "name.givenName",
                          "headerName": "Given Name",
                        },
                        Object {
                          "field": "name.familyName",
                          "headerName": "Family Name",
                        },
                        Object {
                          "field": "gender",
                          "headerName": "Gender",
                        },
                        Object {
                          "field": "age",
                          "headerName": "Age",
                        },
                        Object {
                          "headerName": "Checked-in Time",
                          "valueGetter": [Function],
                        },
                      ]
                    }
                    rowData={Array []}
                    rowSelectedActionCreators={
                      Array [
                        [Function],
                      ]
                    }
                  >
                    <DataGrid
                      columnDefs={
                        Array [
                          Object {
                            "field": "uuid",
                            "headerName": "uuid",
                            "hide": true,
                          },
                          Object {
                            "headerName": "ART",
                            "valueGetter": [Function],
                          },
                          Object {
                            "headerName": "EID",
                            "valueGetter": [Function],
                          },
                          Object {
                            "headerName": "NCD",
                            "valueGetter": [Function],
                          },
                          Object {
                            "field": "name.givenName",
                            "headerName": "Given Name",
                          },
                          Object {
                            "field": "name.familyName",
                            "headerName": "Family Name",
                          },
                          Object {
                            "field": "gender",
                            "headerName": "Gender",
                          },
                          Object {
                            "field": "age",
                            "headerName": "Age",
                          },
                          Object {
                            "headerName": "Checked-in Time",
                            "valueGetter": [Function],
                          },
                        ]
                      }
                      dispatch={[Function]}
                      rowData={Array []}
                      rowSelectedActionCreators={
                        Array [
                          [Function],
                        ]
                      }
                    >
                      <div
                        style={
                          Object {
                            "height": "600px",
                          }
                        }
                      >
                        <label>
                          Search:
                          <input
                            onChange={[Function]}
                            type="text"
                          />
                        </label>
                        <AgGridReact
                          columnDefs={
                            Array [
                              Object {
                                "field": "uuid",
                                "headerName": "uuid",
                                "hide": true,
                              },
                              Object {
                                "headerName": "ART",
                                "valueGetter": [Function],
                              },
                              Object {
                                "headerName": "EID",
                                "valueGetter": [Function],
                              },
                              Object {
                                "headerName": "NCD",
                                "valueGetter": [Function],
                              },
                              Object {
                                "field": "name.givenName",
                                "headerName": "Given Name",
                              },
                              Object {
                                "field": "name.familyName",
                                "headerName": "Family Name",
                              },
                              Object {
                                "field": "gender",
                                "headerName": "Gender",
                              },
                              Object {
                                "field": "age",
                                "headerName": "Age",
                              },
                              Object {
                                "headerName": "Checked-in Time",
                                "valueGetter": [Function],
                              },
                            ]
                          }
                          enableSorting={true}
                          id="omrsGrid"
                          onGridReady={[Function]}
                          onSelectionChanged={[Function]}
                          rowClassRules="rowClassRules"
                          rowData={Array []}
                          rowSelection="single"
                        >
                          <div
                            style={
                              Object {
                                "height": "100%",
                              }
                            }
                          />
                        </AgGridReact>
                      </div>
                    </DataGrid>
                  </Connect(DataGrid)>
                </div>
              </List>
            </div>
          </ScreeningQueue>
        </Connect(ScreeningQueue)>
      </div>
    </BloodPressureQueue>
  </Connect(BloodPressureQueue)>
</Provider>
`;
